back-end/objects/RModelWrapper.py

```diff
@@ -138,18 +122,105 @@ def create_model(self, fields: dict):
         self.db_conn.session.add(model)
         self.db_conn.session.commit()
 
-    def list_models(self) -> Models:
+    def list_models(self) -> list[Models]:
         return Models.query.all()
 
-    def delete_model_by_name(self, name):
-        model_to_delete = Models.query.filter_by(name=name).first()
+    def delete_model_by_name(self, name) -> Models:
+        model_to_delete = Models.query.filter_by(nickname=name).first()
         if model_to_delete:
             db.session.delete(model_to_delete)
             db.session.commit()
+            return model_to_delete
         else:
             print(
                 f"Attempting to delete a model that does not exist. Model name: {name}"
             )
+            return None
+
+    def get_current_model(self):
+        return self.model
+
+    def get_current_model_metadata(self):
+        return self.model_meta_data
+
+    def get_model_by_name(self, name) -> Models:
+        return Models.query.filter_by(nickname=name).first()
+
+    def load_model_by_name(self, model_name: str):
+        model_meta_data = self.get_model_by_name(model_name)
+
+        # TODO: need a way to distinguish between predefined and custom model
+        if model_meta_data.class_name in AVAILABLE_MODELS:
+            model = RModelWrapper.init_pre_defined_model(
+                model_meta_data.class_name,
+                True,
+                model_meta_data,
+                self.num_classes,
```

Comment:
I am thinking that since currently our scheme is one dataset with multiple models, maybe we do not even need the user to input the number of classes when initializing a predefined model because for sure the number of classes should match the dataset they upload. Maybe we can just define `num_classes` in `RDataManager` and use that variable to initialize the predefined model.

Reply:
My current implementation for `UploadModel` api saves `num_classes` used for the predefined model in a '.py' file and regards that as the code file of the predefined model. So I think there are two ways. If we plan to update to a scheme of multiple datasets-multiple models in the future, then saving this variable in a file is reasonable. Otherwise for now, the approach I mentioned above seems to be better.

Reply:
I think we can use `num_classes` passed in from the launcher everywhere for version 0.3.